pkg/kubefedctl/orphaning/orphaning.go

```diff
@@ -0,0 +1,141 @@
+	"sigs.k8s.io/kubefed/pkg/kubefedctl/enable"
+	"sigs.k8s.io/kubefed/pkg/kubefedctl/options"
+	"sigs.k8s.io/kubefed/pkg/kubefedctl/util"
+)
+
+type orphanResource struct {
+	options.GlobalSubcommandOptions
+	typeName          string
+	resourceName      string
+	resourceNamespace string
+}
+
+// Bind adds the join specific arguments to the flagset passed in as an argument.
+func (o *orphanResource) Bind(flags *pflag.FlagSet) error {
+	flags.StringVarP(&o.resourceNamespace, "namespace", "n", "", "If present, the namespace scope for this CLI request")
+	err := flags.MarkHidden("kubefed-namespace")
+	if err != nil {
+		return err
+	}
+	err = flags.MarkHidden("dry-run")
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+// NewCmdOrphaning the head of orphaning-deletion sub commands
+func NewCmdOrphaning(cmdOut io.Writer, config util.FedConfig) *cobra.Command {
+	//	opts := &orphanResource{}
+	cmd := &cobra.Command{
+		Use:   "orphaning-deletion",
+		Short: "Manage orphaning delete policy",
+		Long:  "Manage orphaning delete policy",
+		Run: func(cmd *cobra.Command, args []string) {
+			err := cmd.Help()
+			if err != nil {
+				klog.Fatalf("Error: %v", err)
+			}
+		},
+	}
+	cmd.AddCommand(newCmdEnableOrphaning(cmdOut, config))
+	cmd.AddCommand(newCmdDisableOrphaning(cmdOut, config))
+	cmd.AddCommand(newCmdStatusOrphaning(cmdOut, config))
+
+	return cmd
+}
+
+//  Complete ensures that options are valid and marshals them if necessary.
+func (o *orphanResource) Complete(args []string, config util.FedConfig) error {
+
+	if len(args) == 0 {
+		return errors.New("resource type is required")
+	}
+
+	o.typeName = args[0]
+
+	if len(args) == 1 {
+		return errors.New("resource name is required")
+	}
+	o.resourceName = args[1]
+
+	return o.SetNamespace(config)
+}
+
+// If namespace was not set as an argument, it will be the current context namespace
+func (o *orphanResource) SetNamespace(config util.FedConfig) error {
+
+	if o.resourceNamespace != "" {
+		return nil
+	}
+	ns, _, err := config.GetClientConfig(o.HostClusterContext, o.Kubeconfig).Namespace()
+	if err != nil {
+		return errors.Wrapf(err, "Failed to get ClientConfig for host cluster context '%s' and kubeconfig '%s'", o.HostClusterContext, o.Kubeconfig)
+	}
+
+	if ns == "" {
+		ns = "default"
+	}
+	o.resourceNamespace = ns
+	return nil
+}
+
+// Returns the Federated resource where the orphaning-deletion will be managed, and a corresponding Resources Interface
+func (o *orphanResource) GetFederatedResource(config util.FedConfig) (*unstructured.Unstructured, dynamic.ResourceInterface, error) {
+	hostConfig, err := config.HostConfig(o.HostClusterContext, o.Kubeconfig)
+	if err != nil {
+		return nil, nil, errors.Wrap(err, "Failed to get host cluster config")
+	}
+	// Lookup kubernetes API availability
+	apiResource, err := enable.LookupAPIResource(hostConfig, o.typeName, "")
+	if err != nil {
+		return nil, nil, errors.Wrapf(err, "Failed to find target API resource %s", o.typeName)
```

Comment:
I think the error message here should reflect a failure to find a type corresponding to the provided type name rather than potentially confusing the user with mention of an api resource.

Reply:
updated the error message